lib/BuildSystem/BuildSystem.cpp

```diff
@@ -422,12 +416,16 @@ class TargetTask : public Task {
       return;
     }
 
-    if (hasMissingInput) {
+    if (namesOfMissingInputs.size() > 0) {
+      std::ostringstream inputs;
+      std::copy(namesOfMissingInputs.begin(), namesOfMissingInputs.end()-1, std::ostream_iterator<std::string>(inputs, ", "));
```

Comment:
I think it is more readable to just use a regular for loop rather than `std::copy`.